tests/cpp/test_matmul_aten_evaluation.cpp

```diff
@@ -409,6 +410,96 @@ TEST_F(MatmulATenEvaluationTest, LinearWithBias) {
+
+  if (A->nDims() == 1 && B->nDims() == 1) {
+    // [K] @ [K] = []
+    // Note there is no IterType::Reduction dim ever in this case because we
+    // translate to a mul+sum+cast
+    EXPECT_EQ(output->nDims(), 0);
+    // When K is Broadcast, we squeeze then multiply then cast instead
+    if (!k_bcast) {
+      EXPECT_TRUE(checkMapped(A->axis(0), B->axis(0))); // K
+    }
+  } else if (A->nDims() > 1 && B->nDims() == 1) {
+    // [..., iM, iK] @ [iK] = [..., iM, rK]
+    ASSERT_EQ(output->nDims(), A->nDims() + red_dims - 1);
+    EXPECT_TRUE(checkMapped(A->axis(-2), output->axis(-1 - red_dims))); // M
+    if (!k_bcast) {
+      EXPECT_TRUE(checkMapped(A->axis(-1), B->axis(0))); // K
+      EXPECT_TRUE(checkMapped(A->axis(-1), output->axis(-1))); // K
+    }
+    // Check that batch dims are mapped
+    for (int64_t i : c10::irange(output->nDims() - red_dims - 1)) {
+      if (!A->axis(i)->isBroadcast()) {
+        EXPECT_TRUE(checkMapped(A->axis(i), output->axis(i)));
+      }
+    }
+  } else if (A->nDims() == 1 && B->nDims() > 1) {
+    // [iK] @ [..., iK, iN] = [..., iN, rK]
+    ASSERT_EQ(output->nDims(), B->nDims() + red_dims - 1);
+    EXPECT_TRUE(checkMapped(B->axis(-1), output->axis(-1 - red_dims))); // N
+    if (!k_bcast) {
+      EXPECT_TRUE(checkMapped(A->axis(0), B->axis(-2))); // K
+      EXPECT_TRUE(checkMapped(A->axis(0), output->axis(-1))); // K
+    }
+    // Check that batch dims are mapped
+    for (int64_t i : c10::irange(output->nDims() - red_dims - 1)) {
+      if (!B->axis(i)->isBroadcast()) {
+        EXPECT_TRUE(checkMapped(B->axis(i), output->axis(i)));
+      }
+    }
+  } else if (A->nDims() > 1 && B->nDims() > 1) {
+    // [..., iM, iK] @ [..., iK, iN] = [..., iM, iN, rK]
+    ASSERT_EQ(output->nDims(), std::max(A->nDims(), B->nDims()) + red_dims);
+    EXPECT_TRUE(checkMapped(A->axis(-2), output->axis(-2 - red_dims))); // M
+    EXPECT_TRUE(checkMapped(B->axis(-1), output->axis(-1 - red_dims))); // N
+    if (!k_bcast) {
+      EXPECT_TRUE(checkMapped(A->axis(-1), B->axis(-2))); // K
+      EXPECT_TRUE(checkMapped(A->axis(-1), output->axis(-1))); // K
+    }
+    // Check that batch dims are mapped
+    // Note that A and B can have different dimensions, so here we count
+    // backwards from the innermost batch dimension. Then we check that the axis
+    // exists (is not negative) and is not Broadcast before checking mapping.
+    for (int64_t i : c10::irange(output->nDims() - red_dims - 2)) {
+      int64_t i_a = A->nDims() - 3 - i;
+      int64_t i_b = B->nDims() - 3 - i;
+      int64_t i_out = output->nDims() - red_dims - 3 - i;
+      if (i_a >= 0 && !A->axis(i_a)->isBroadcast()) {
+        EXPECT_TRUE(checkMapped(A->axis(i_a), output->axis(i_out)));
+      }
+      if (i_b >= 0 && !B->axis(i_b)->isBroadcast()) {
+        EXPECT_TRUE(checkMapped(B->axis(i_b), output->axis(i_out)));
+      }
+    }
+  } else {
+    std::cerr << "Unhandled set of input dimensions" << std::endl;
```

Comment:
nit: Consider using `NVF_ERROR` here.

Reply:
I wanted to use `EXPECT_TRUE` to prevent stopping the test when we encounter this condition. But `EXPECT_*` does not accept any additional error message. We don't really need to print this though; I could just place it in a comment then the test would still fail if we had some new cases that weren't handled and we would see the comment when we investigate.

Reply:
Oh right, `EXPECT_TRUE` does not abort. We can stick with this then.